cloudinit/distros/__init__.py

```diff
@@ -1006,14 +1006,20 @@ def write_doas_rules(self, user, rules, doas_file=None):
                     )
                     raise e
 
-    def ensure_sudo_dir(self, path, sudo_base="/etc/sudoers"):
+    def ensure_sudo_dir(
+            self, path, sudo_base=("/etc/sudoers","/usr/etc/sudoers")):
         # Ensure the dir is included and that
         # it actually exists as a directory
         sudoers_contents = ""
         base_exists = False
-        if os.path.exists(sudo_base):
-            sudoers_contents = util.load_text_file(sudo_base)
-            base_exists = True
+        admin_sudo_base = "/etc/sudoers"
+        if not isinstance(sudo_base, (list, tuple)):
+            sudo_base = [sudo_base]
+        for sudoers_base in sudo_base:
+            if os.path.exists(sudoers_base):
```

Comment:
> Will there be a case where /usr/etc/sudoers exists bit /etc/sudoers does not? It seems this is possible given [this read-only FS note in SUSE docs which talk about visudo writing to /etc/sudoers if /usr/etc/sudoers exists but /etc/sudoers does not](https://documentation.suse.com/smart/systems-management/html/sudo-configure-superuser-privileges/index.html).
> 

Yes, in newer SUSE distributions `/etc/sudoers` will not exist by default.

> Do we want to completely ignore /usr/etc/sudoers in any environment which already has an /etc/sudoers file? There is no active merge of /usr/etc/sudoers content with primary /etc/sudoers anywhere.

If `/etc/sudoers` exists the underlying premise is that the user (sysadmin) has made the modifications they want to have in place. Should the `include` directive be missing we add it to this file to match the current behavior. If `/etc/usr/sudoers` exists and contains the `include` directive and we would not add the `include` directive to an existing `/etc/sudoers` we would change the behavior compared to the current implementation because the include files would be processed at a different point in time.